module.js

```diff
@@ -0,0 +1,66 @@
+var ExtensionUtils = imports.misc.extensionUtils;
+var Extension = ExtensionUtils.getCurrentExtension();
+
+function ExtensionModule() {
+    return Extension.imports.extension;
+}
```

Comment:
I understand the reason for doing it this way. But it feels very hacky.

Ideally I would like to reorganize the code to get rid of the dependency cycles, rather than doing it this way. But that can also be done later.

Also I have never run into the dependency cycle problem in a JS project before, although maybe I just got lucky^^

Reply:
Hey @Lythenas, yeah, you run into it when trying to refactor (especially to remove `init` methods) - e.g. previously we were statically defining some vars - which other modules dependended on being defined before they ran `enable`.  Problem was, when moving away fron `init` completely broke then... hence, I needed to play around with alot with ordering of `modules` const in `extension.js`.  For examples, `Tiling` had to be before `Settings.js`, but other modules needed `Settings.js` before their enable could be called...

In any case, it was pretty much `dependency hell` until I got it just right... only to be broken again when soon after by removing statically defined vars (object creations - which is the same as calling `init`).

The current approach solves these and means we no longer depend on the module `enable` method being called in specific order.

Reply:
It's pretty much the internal module pattern - modelled it after some ideas in here: https://medium.com/visual-development/how-to-fix-nasty-circular-dependency-issues-once-and-for-all-in-javascript-typescript-a04c987cf0de so pretty much the internal module patter

But, I'd happily accept a PR looking at other/cleaner approaches here!

Reply:
I would leave it like this for now. If we get time, we can refactor it in the future. But it might require a lot of refactoring^^